test/language/expressions/does-not-equals/bigint-and-object.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2017 Josh Wolfe. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Non-strict inequality comparison of BigInt values and non-primitive objects
+esid: sec-abstract-equality-comparison
+info: |
+  10. If Type(x) is either String, Number, BigInt, or Symbol and Type(y) is Object, return the result of the comparison x == ? ToPrimitive(y).
+  11. If Type(x) is Object and Type(y) is either String, Number, BigInt, or Symbol, return the result of the comparison ? ToPrimitive(x) == y.
+
+  then after the recursion:
+
+  1. If Type(x) is the same as Type(y), then
+    a. Return the result of performing Strict Equality Comparison x === y.
+  ...
+  6. If Type(x) is BigInt and Type(y) is String,
+    a. Let n be StringToBigInt(y).
+    b. If n is NaN, return false.
+    c. Return the result of x == n.
+  7. If Type(x) is String and Type(y) is BigInt, return the result of y == x.
+
+features: [BigInt]
+---*/
+
+assert.sameValue(0n != {}, true);
+assert.sameValue({} != 0n, true);
```

Comment:
these tests can also use comparisons for objects with a `[[BigIntData]]` internal slot.